### PR TITLE
Update export step5 for strata

### PR DESCRIPTION
**Pull request by juruchi**
strata_id for DTH (#59 #60 #61) and DTHRED (#62 #63 #64) datasets are programmatically filled in due to the large number of strata_id variables. Rather than reviewing the by_var_strata codebook sheet, Goro would prefer to review the table that is used to fill in the strata_id variable i.e the table that shows raw, name, and values for all available strata. This table would be available in the base version of the codebook, and would be removed prior to exporting the final codebook.

Here we:

1. Update ```etl_DTH1_v1.0.R``` to include the additional strata sheet in the base codebook
2. Update ```etl_export_step5.R``` so all validation tests are passed for DTH and DTHRED cases
3. Write updated base codebooks with the additional strata sheet for DTH1

### CODE REVIEW

**Comment by ran-codes** [comment]
thanks @juruchi will take a look tomorrow morning.

**Comment by ran-codes** [comment]
# PR review notes for Ran

## Review Action Items

- [ ] Update etl_DTH1_v1.0.R to include the additional strata sheet in the base codebook
- [ ] Update etl_export_step5.R so all validation tests are passed for DTH and DTHRED cases
- [ ] Write updated base codebooks with the additional strata sheet for DTH1

File: R/_etl/etl_export_step5.R [hunk 1]
```diff
@@ -101,42 +105,50 @@ etl_export_step5 = function(dataset_id_tmp, stage, active__codebook, i){
...
+        }) %>%
         all()
       if (!valid_no_commas) cli_alert_danger("valid_no_commas ERROR: commas found in base cdbk tables")
     }
     
     { ## `valid_tables` ------------------------------------------------------------
       tables_missing =  names(templates) %>% keep(~!.x%in% names(active__codebook))
-      tables_extra =  names(active__codebook) %>% keep(~!.x%in% names(templates))
+      
+      if (dataset_id_tmp %in% c("DTH1", "DTH2", "DTH3", "DTHRED1", "DTHRED2", "DTHRED3") & stage == "base"){
```
Comment (ran-codes) [inline R/_etl/etl_export_step5.R:144]: do you think this strata_contriubction work flow will be applicable to other data? just a note for future if this is a more common workflow we can add this as an argument into the elt_export_step() function so these don't have to be hard coded.

lets keep as is but for future!

File: R/_etl/etl_export_step5.R [hunk 2]
```diff
@@ -101,42 +105,50 @@ etl_export_step5 = function(dataset_id_tmp, stage, active__codebook, i){
...
-      
+
+
       if (!valid_pk){
         cli_alert_danger('_src `valid_pk` ERROR: table name and primary keys in src tables are inconsistent. Please check!')
       }
     }
     { ## `valid_no_commas` ------------------------------------------------------------
-      valid_no_commas = active__codebook %>% 
+      valid_no_commas = active__codebook %>%
```
Comment (ran-codes) [inline R/_etl/etl_export_step5.R:117]: these are all just white space changes. ntoe to self

**Comment by ran-codes** [comment]
great job @juruchi ! merging!

**Comment by juruchi** [comment]
@ran-codes 
>  moved the crosswalks into the git repo... we should eventually have the code that generates this under /scripts to have it under version control.

I talked with Goro, and there's a different file (from the unencrypted folder) that should be used for all three tiers. There's some cleaning that has to be done to the file before it can be used. Where is the best place to create a script to read in and clean this file?

**Comment by ran-codes** [comment]
> different file that should be used for all three tiers. 

Can you double check the right imports accordingly in each of the reno funcs?


> Where is best place?

Not sure yet. can I just scan the code quickly.  You can just create a folder in /code/scratch (see commit above) to store that codebase and we can decide then where to put it. 

But in general. code and metadata should be on GitHub... either this repo or another. Version control is huge. Data can stay on server.

**Comment by juruchi** [comment]
@ran-codes The cleaning script for the GHE code/name crosswalk has been saved in /code/search and I updated the six renovation scripts accordingly.

**Pull request merged**